crates/storage_models/src/query/generics.rs

```diff
@@ -359,6 +359,37 @@ where
     .attach_printable_lazy(|| "Error filtering records by predicate")
 }
 
+#[instrument(level = "DEBUG", skip_all)]
+pub(super) async fn generic_filter_order<T, P, R, Expr>(
+    conn: &PgPooledConn,
+    predicate: P,
+    limit: Option<i64>,
+    expr: Expr,
+) -> StorageResult<Vec<R>>
+where
+    Expr: diesel::Expression,
+    T: FilterDsl<P> + HasTable<Table = T> + Table + 'static,
+    <T as FilterDsl<P>>::Output: OrderDsl<Expr> + Send + 'static,
+    <<T as FilterDsl<P>>::Output as OrderDsl<Expr>>::Output: LimitDsl + Send + 'static,
+    <<<T as FilterDsl<P>>::Output as OrderDsl<Expr>>::Output as LimitDsl>::Output:
+        LoadQuery<'static, PgConnection, R> + QueryFragment<Pg> + Send,
+    R: Send + 'static,
+{
+    let query = <T as HasTable>::table()
+        .filter(predicate)
+        .order(expr)
+        .limit(limit.unwrap_or(100));
```

Comment:
Let me know if the default value has to be added as a const in the beginning of the file or as a config.

Reply:
Since limit is passed by caller, You can just move it to CONST.